cognite/client/data_classes/agents/chat.py

```diff
@@ -85,6 +86,194 @@ def _load_content(cls, data: dict[str, Any]) -> UnknownContent:
 }
 
 
+@dataclass
+class Action(CogniteObject, ABC):
+    """Base class for all action types that can be provided to an agent."""
+
+    _type: ClassVar[str]
+
+    @classmethod
+    def _load(cls, data: dict[str, Any], cognite_client: CogniteClient | None = None) -> Action:
+        """Dispatch to the correct concrete action class based on `type`."""
+        action_type = data.get("type", "")
+        action_class = _ACTION_CLS_BY_TYPE.get(action_type, UnknownAction)
+        return action_class._load_action(data, cognite_client)
+
+    @abstractmethod
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        """Dump the action to a dictionary."""
+        ...
+
+    @classmethod
+    @abstractmethod
+    def _load_action(cls, data: dict[str, Any], cognite_client: CogniteClient | None = None) -> Action:
+        """Create a concrete action instance from raw data."""
+        ...
+
+
+@dataclass
+class ClientToolAction(Action):
+    """A client-side tool definition that can be called by the agent.
+
+    Args:
+        name (str): The name of the client tool to call.
+        description (str): A description of what the function does. The language model will use this description when selecting the function and interpreting its parameters.
+        parameters (dict[str, Any]): The parameters the function accepts, described as a JSON Schema object.
+    """
+
+    _type: ClassVar[str] = "clientTool"
+    name: str
+    description: str
+    parameters: dict[str, Any]
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        return {
+            "type": self._type,
+            "clientTool": {
```

Comment:
I just not like strings if I can avoid them, so easy to get typos.
```suggestion

    def dump(self, camel_case: bool = True) -> dict[str, Any]:
        return {
            "type": self._type,
            self._type: {
                "name": self.name,
```